### PR TITLE
various fixes for building with shared libraries

### PR DESCRIPTION
**Pull request by truemedian**
also adds a CI check for building with all shared libraries to ensure they don't regress

Most of these regressions were introduced in #285 as a result of not being covered by CI

Primarily focused on fixing #280

### CODE REVIEW

**Comment by Bilal2453** [comment]
A Luvi release is blocked until this is merged.

Seems mostly good to me, be it some parts are confusing as I don't fully understand the build scripts.

@squeek502 / @zhaozg.

**Review by squeek502 — APPROVED** [review]

**Pull request merged**
